cpu/saml21/cpu.c

```diff
@@ -82,6 +90,37 @@ uint32_t sam0_gclk_freq(uint8_t id)
     }
 }
 
+void cpu_pm_cb_enter(int deep)
+{
+    if (!deep) {
+        return;
+    }
+
+    /* If you are using saml21 rev. B, switch Main Clock to OSCULP32 during standby
+       to work around errata 1.2.1.
+       See discussion in #13441  */
+    assert((DSU->DID.bit.REVISION > 1) || ((PM->STDBYCFG.reg & 0x80) == 0));
+
+    /* errata 51.1.5 – When VDDCORE is supplied by the BUCK converter in performance
+                       level 0, the chip cannot wake-up from standby mode because the
+                       VCORERDY status is stuck at 0. */
+    if (USE_VREG_BUCK) {
+        sam0_set_voltage_regulator(SAM0_VREG_LDO);
+    }
```

Comment:
SAML21 is set to PL2 in cpu_init, so I guess  it would be better to replace it by
```suggestion
    /* errata 51.1.5 – When VDDCORE is supplied by the BUCK converter in performance
                       level 0, the chip cannot wake-up from standby mode because the
                       VCORERDY status is stuck at 0. */
    if (USE_VREG_BUCK && !PM->PLCFG.bit.PLSEL) {
        sam0_set_voltage_regulator(SAM0_VREG_LDO);
    }
```